tests/test_materials_library.py

```diff
@@ -79,11 +101,80 @@ def test_against_nuc_data():
         metadata={"name": "Plutonium Bromide"}).expand_elements()
 
     obs_gasoline = obs_matslib["Gasoline"]
-    assert_equal(set(obs_gasoline.comp.items()), set(gasoline.comp.items()))
+    # remove empty elements
+    assert_equal( { (x,val) for x,val in set(obs_gasoline.comp.items()) if val > 0 }, 
+            set(gasoline.comp.items()))
     assert_equal(obs_gasoline.density, gasoline.density)
     assert_equal(obs_gasoline.metadata["name"], gasoline.metadata["name"])
 
     obs_pubr3 = obs_matslib["Plutonium Bromide"]
-    assert_equal(set(obs_pubr3.comp.items()), set(pubr3.comp.items()))
+    # remove empty elements
+    assert_equal( { (x,val) for x,val in set(obs_pubr3.comp.items()) if val > 0 }, 
+            set(pubr3.comp.items()))
     assert_equal(obs_pubr3.density, pubr3.density)
     assert_equal(obs_pubr3.metadata["name"], pubr3.metadata["name"])
+
+def test_matlib_json():
+    filename = "matlib.json"
+    water = Material()
+    water.from_atom_frac({10000000: 2.0, 80000000: 1.0})
+    water.metadata["name"] = "Aqua sera."
+    lib = {"leu": Material(leu), "nucvec": nucvec, "aqua": water}
+    wmatlib = MaterialLibrary(lib)
+    wmatlib.write_json(filename)
+    
+    rmatlib = MaterialLibrary()
+    rmatlib.from_json(filename)
+    assert_equal(set(wmatlib), set(rmatlib))
+    for key in rmatlib:
+        assert_mat_almost_equal(wmatlib[key], rmatlib[key])
+    os.remove(filename)
+
+def test_matlib_hdf5_nuc_data():
+    matlib = MaterialLibrary()
+    matlib.from_hdf5(nuc_data, datapath="/material_library/materials")
+    matlib.write_hdf5("matlib_test.h5", "/materials")
+    mat_lib_load_test = MaterialLibrary("matlib_test.h5", "/materials")
```

Comment:
Should you test that this worked?

Reply:
I guess we can add some test
(was not present before: https://github.com/pyne/pyne/blob/4de6dfcebb0022fd855c90ccc170be0854720c01/tests/test_material.py#L1786)

Reply:
added one to ensure `matlib` and `mat_lib_load_test` are identical